### ISSUE TITLE
Priority setting on exporters/importers doesn't seem to function

### ISSUE THREAD

**Issue by Jack-McKalling** [CyclopsMC/IntegratedDynamics#1405]
#### Issue type:

- :bug: Bug 

____

#### Short description:

When setting a priority value on an exporter/importer, one would expect the one with the highest priority (on the same network) to function before the others. The current implementation works exactly as well as not having any priority setting.

![integrated tunnels export priorities](https://github.com/user-attachments/assets/0299c0dc-3408-423e-8761-d9534b518b9c)
This setup will show varying results with uneven stack distribution in both targets, irrespective of the priority setting.

#### Steps to reproduce the problem:

1. Build the above setup, with both exporters set to transfer 1 item per tick (so you can test with just 1 stack of items)
2. Set the one exporter to a different priority than the other
3. Insert a full stack of any stackable item in the input chest on the left
4. Observe how the stack is split across both destinations with varying counts

#### Expected behaviour:

Since both exporters are on the same network, but have different priorities, I would expect one exporter to _always_ be prioritized on each tick, and that _all items_ exported end up at the destination with the highest priority.

I would expect that a lower priority destination would only be used if:
- The higher priorities are full
- The higher priorities don't accept the transferred item anymore (e.g. when the storage is full of one item but can still accept others)
- The higher priorities have a lower tick rate (e.g. this one exports each tick but the others only once every 4 ticks), then the priorities should be resolved according to the ticks they're active in
- The higher priorities have further stack requirements that this one doesn't have
- etc, all obvious stuff

____

#### Versions:

- Dynamics: 1.23.9
- Tunnels: 1.8.28
- Core: 1.25.1-630
- CC: 2.9.5
- Minecraft: 1.21.1
- Mod loader version:  Neoforge 21.1.69

**Comment by rubensworks** [comment]
Thanks for reporting!

**Comment by Jack-McKalling** [comment]
I'm respecting that the priority setting does probably have some kind of effect somewhere. But the above is the _interpretation of the player_, and the most intuitive behaviour. Any other behaviour should be explained, as relying on this setting will have unexpected results.

For instance if they want to sort between two destinations, where one is a safe-keeping destination for the same items as the other, and they want to make sure they always have the former stocked before the other (e.g. a trashcan overflow system). With a setup like this, that can easily result in the potentially 100% loss of all ingredients.
